aten/src/ATen/core/boxing/impl/make_boxed_from_unboxed_functor.h

```diff
@@ -230,6 +230,22 @@ using supported_primitive_arg_types = guts::typelist::typelist<
       return ivalue_to_arg<std::vector<T>, AllowDeprecatedTypes>::call(std::move(v));
     }
   };
+  template<bool AllowDeprecatedTypes>
+  struct ivalue_to_arg<optional<ArrayRef<int64_t>>, AllowDeprecatedTypes> final {
+    // If an argument is optional<ArrayRef<int64_t>>, convert the IValue to a optional<std::vector<int64_t>> and pass that
+    // to the operator.
```

Comment:
Uhh, so I guess you're just following the same pattern as the code above (which was already doing this), but why do we have to convert the ivalue into a `std::vector<int64_t>`. Isn't this some totally unnecessary allocation?

Reply:
This unboxes an `IValue` (which is a list, so it holds a `std::vector<IValue>`) to an `ArrayRef<int64_t>`. The only way of doing that is to copy the full list. I think I mentioned this a few times when I argued for not allowing `ArrayRef` in kernel signatures, but the resolution was that we said perf doesn't matter here.